crates/core/component/stake/src/component/validator_handler/validator_store.rs

```diff
@@ -227,23 +227,6 @@ pub trait ValidatorDataRead: StateRead {
             .map_ok(|opt: Option<Validator>| opt.map(|v: Validator| v.consensus_key))
             .boxed()
     }
-
-    /// Returns a list of **all** known validators metadata.
-    async fn validator_definitions(&self) -> Result<Vec<Validator>> {
-        self.prefix(state_key::validators::definitions::prefix())
-            .map_ok(|(_key, validator)| validator)
-            .try_collect()
-            .await
-    }
-
-    /// Returns a list of **all** known validators identity keys.
-    async fn validator_identity_keys(&self) -> Result<Vec<IdentityKey>> {
-        self.prefix(state_key::validators::definitions::prefix())
-            .map_ok(|(_key, validator)| validator)
-            .map_ok(|validator: Validator| validator.identity_key)
-            .try_collect()
-            .await
-    }
```

Comment:
these are removed, and now live in a place where they are explicitly only for use in tests. that'll help prevent future regressions, but keeps the friendly ergonomics around for test assertions' sake.

Reply:
great